### PR TITLE
Minor cleanup, no functional changes

### PR DESCRIPTION
**Pull request by hmusum**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java [hunk 1]
```diff
@@ -41,49 +41,38 @@ public FilesApplicationFile(Path path, File file) {
...
-            if (path.isDirectory()) {
-                File[] files = path.listFiles();
-                for(int i=0; i<files.length; i++) {
-                    if(files[i].isDirectory()) {
-                        deleteFile(files[i]);
-                    } else {
-                        files[i].delete();
-                    }
-                }
+        if (path.exists() && path.isDirectory()) {
```
Comment (hakonhall) [inline config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java:69]: isDirectory => exists, so remove `path.exists()` ?

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java [hunk 2]
```diff
@@ -41,49 +41,38 @@ public FilesApplicationFile(Path path, File file) {
...
     public ApplicationFile delete() {
-        log.log(Level.FINE, () -> "Delete " + file);
-        if (file.isDirectory() && !listFiles().isEmpty()) {
-            throw new RuntimeException("files. Can't delete, directory not empty: " + this  + "(" + listFiles() + ")." + listFiles().size());
-        }
-        if (file.isDirectory() && file.listFiles() != null && file.listFiles().length > 0) {
-            for (File f : file.listFiles()) {
-                deleteFile(f);
+        if (file.isDirectory()) {
+            if (!listFiles().isEmpty())
```
Comment (hakonhall) [inline config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java:52]: I believe this `delete()` can only be invoked on a directory if the directory only has entries that start with `.`, and in that case any such file and directory-trees are deleted.

This seems strange to me - does it make sense to you?

It is possibly related to the fact that this class creates a .meta directory after deletion.

**Review by hakonhall — DISMISSED** [review]

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java (hunk 1 above)
Comment (hmusum) [reply config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java:69]: Fixed

File: config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java (hunk 2 above)
Comment (hmusum) [reply config-application-package/src/main/java/com/yahoo/config/model/application/provider/FilesApplicationFile.java:53]: Discussed offline, added some TODOs

**Pull request merged**